pkg/tsdb/azuremonitor/azure-resource-graph-datasource.go

```diff
@@ -0,0 +1,257 @@
+// 1. builds the AzureMonitor url and querystring for each query
+// 2. executes each query by calling the Azure Monitor API
+// 3. parses the responses for each query into the timeseries format
+func (e *AzureResourceGraphDatasource) executeTimeSeriesQuery(ctx context.Context, originalQueries []plugins.DataSubQuery,
+	timeRange plugins.DataTimeRange) (backend.QueryDataResponse, error) {
+	result := backend.QueryDataResponse{
+		Responses: map[string]backend.DataResponse{},
+	}
+
+	queries, err := e.buildQueries(originalQueries, timeRange)
+	if err != nil {
+		return backend.QueryDataResponse{}, err
+	}
+
+	for _, query := range queries {
+		result.Responses[query.RefID] = e.executeQuery(ctx, query, timeRange)
+	}
+
+	return result, nil
+}
+
+func (e *AzureResourceGraphDatasource) buildQueries(queries []plugins.DataSubQuery,
+	timeRange plugins.DataTimeRange) ([]*AzureResourceGraphQuery, error) {
+	var azureResourceGraphQueries []*AzureResourceGraphQuery
+
+	for _, query := range queries {
+		queryBytes, err := query.Model.Encode()
+		if err != nil {
+			return nil, fmt.Errorf("failed to re-encode the Azure Resource Graph query into JSON: %w", err)
+		}
+
+		queryJSONModel := argJSONQuery{}
+		err = json.Unmarshal(queryBytes, &queryJSONModel)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode the Azure Resource Graph query object from JSON: %w", err)
+		}
+
+		azureResourceGraphTarget := queryJSONModel.AzureResourceGraph
+		azlog.Debug("AzureResourceGraph", "target", azureResourceGraphTarget)
+
+		resultFormat := azureResourceGraphTarget.ResultFormat
+		if resultFormat == "" {
+			resultFormat = "table"
+		}
+
+		params := url.Values{}
+		rawQuery, err := KqlInterpolate(query, timeRange, azureResourceGraphTarget.Query)
+
+		if err != nil {
+			return nil, err
+		}
+		params.Add("query", rawQuery)
+
+		azureResourceGraphQueries = append(azureResourceGraphQueries, &AzureResourceGraphQuery{
+			RefID:        query.RefID,
+			ResultFormat: resultFormat,
+			Model:        query.Model,
+			Params:       params,
+			Target:       params.Encode(),
+		})
+	}
+
+	return azureResourceGraphQueries, nil
+}
+
+func (e *AzureResourceGraphDatasource) executeQuery(ctx context.Context, query *AzureResourceGraphQuery,
+	timeRange plugins.DataTimeRange) backend.DataResponse {
+	queryResult := backend.DataResponse{}
+
+	query.Params.Add("api-version", argAPIVersion)
+
+	queryResultErrorWithExecuted := func(err error) backend.DataResponse {
+		queryResult = backend.DataResponse{Error: err}
+		frames := data.Frames{
+			&data.Frame{
+				RefID: query.RefID,
+				Meta: &data.FrameMeta{
+					ExecutedQueryString: query.Params.Get("query"),
+				},
+			},
+		}
+		queryResult.Frames = frames
+		return queryResult
+	}
+
+	reqBody := fmt.Sprintf("{'subscriptions':['%s'], 'query':'%s'}",
+		strings.Join(query.Model.Get("subscriptions").MustStringArray(), "','"),
+		query.Model.Get("azureResourceGraph").Get("query").MustString())
```

Comment:
I tried `"subscriptions": fmt.Sprintf("['%s']", strings.Join(query.Model.Get("subscriptions").MustStringArray(), "','")),`, but got `"subscriptions":"['f7152080-b4e8-47ee-9c85-7f1d0e6b72dc','4dc2cd39-7a89-43d8-bebe-8bb501359891']"`. This won't work because `subscriptions` needs to be an array instead of a string. `json.Marshal` converts everything into a string.

Reply:
It works if you just skip the strings.Join. I pushed a change!